lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
+        received_with_arg_list(*with_args)
+      end
+
+      def received_with_arg_list(*args)
+        list = args.map do |a, i|
+          times_count = args.size > 1 ? '(' + times(i) + ')' : ''
+          ["(#{received_arg_list(*a)})", "#{times_count}"].join(" ")
```

Comment:
That's a good point.  I think it would be nice to include `(1 time)` in that case.  Maybe we can exclude the "n times" bit if it was exactly 1 time for each?